### ISSUE TITLE
[Bug]: "ERROR: refresh window too small" on a refresh_continuous_aggregate that stacks two continuous aggregates

### ISSUE THREAD

**Issue by srieding** [timescale/timescaledb#5759]
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Continuous aggregate

### What happened?

When I call refresh_continuous_aggregate, with exactly one bucket size (1 month) from window_start till window_end it works if the continuous aggregate is defined straight on the base table but it does not work if there is another continuous aggregate in between with a smaller bucket size (1day). 

### TimescaleDB version affected

2.10.2

### PostgreSQL version used

15.1

### What operating system did you use?

window 11 x64

### What installation method did you use?

Docker

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
[22023] ERROR: refresh window too small Detail: The refresh window must cover at least one bucket of data. Hint: Align the refresh window with the bucket time zone or use at least two buckets.
```

### How can we reproduce the bug?

```bash
-- starting on an empty database
-- create a base (hyper)table
CREATE TABLE measurements (
                              "generated_at" timestamptz not null,
                              "value" int4 not null,
                              "location_id" int4 not null
);

ALTER TABLE "measurements" ADD CONSTRAINT measurements_pkey PRIMARY KEY (location_id, generated_at);

SELECT create_hypertable (
               'measurements',
               'generated_at',
               chunk_time_interval => INTERVAL '7 days',
               if_not_exists => TRUE,
               create_default_indexes => FALSE
           );

-- Add a continuous aggregate straight on the base table.
CREATE MATERIALIZED VIEW summary_month_on_base WITH (timescaledb.continuous)
AS SELECT
       time_bucket('1 month', measurements.generated_at::timestamptz, 'UTC') AS "time",
       measurements.location_id,
       count(*) ::integer AS count
   FROM measurements
   GROUP BY measurements.location_id, (time_bucket('1 month'::interval, measurements.generated_at::timestamptz, 'UTC'))
WITH NO DATA;

-- Add a continuous aggregate on a continuous aggregate.
CREATE MATERIALIZED VIEW summary_day WITH (timescaledb.continuous)
AS SELECT
       time_bucket('1 day', measurements.generated_at::timestamptz, 'UTC') AS "time",
       measurements.location_id,
       count(*) AS count
   FROM measurements
   GROUP BY measurements.location_id, (time_bucket('1 day'::interval, measurements.generated_at::timestamptz, 'UTC'))
WITH NO DATA;

CREATE MATERIALIZED VIEW summary_month WITH (timescaledb.continuous)
AS SELECT
       time_bucket('1 month', summary_day.time::timestamptz, 'UTC') AS "time",
       summary_day.location_id,
       sum(summary_day.count)::integer AS count
   FROM summary_day
   GROUP BY summary_day.location_id, (time_bucket('1 month'::interval, summary_day.time::timestamptz, 'UTC'))
WITH NO DATA;

-- This one fails
CALL refresh_continuous_aggregate('summary_month', '2022-12-01T00:00:00.000Z'::timestamptz, '2023-01-01T00:00:00.000Z'::timestamptz);
-- This one works
CALL refresh_continuous_aggregate('summary_month_on_base', '2022-12-01T00:00:00.000Z'::timestamptz, '2023-01-01T00:00:00.000Z'::timestamptz);
-- This returns UTC
show timezone;
```

**Comment by jnidzwetzki** [comment]
Hello @srieding,

Thanks for bringing this to our attention and for the detailed steps to reproduce the problem. I was able to reproduce the inconsistent behavior in my local environment. However, by adding a day to the refresh interval, I could perform a refresh. Could you verify if the following query works as a workaround in your environment?

```sql
CALL refresh_continuous_aggregate('summary_month_on_base', '2022-12-01T00:00:00.000Z'::timestamptz, '2023-01-02T00:00:00.000UTC'::timestamptz);
```

**Comment by srieding** [comment]
Yes the workaround works for me. That is what I'm doing at the moment.

**Comment by linusback** [comment]
I'm experiencing the same bug, but it is not solved able to just add 1 day to the refresh interval. 
Sometimes i have to add 5,6 or more days. Further more it doesn't seem to always populate the data. Sometimes it runs but the bucket is empty. I hope it is okey that i write here instead of making a new issue. 
I have modified the repruduction script to show the problems.

### TimescaleDB version affected
2.11.2

### PostgreSQL version used
13.12

### What operating system did you use?
Ubuntu 22.04.3 x64

### What installation method did you use?
Docker with image timescale/timescaledb:2.11.2-pg13

### What platform did you run on?
On prem/Self-hosted

### How can we reproduce the bug?
`
-- starting on an empty database
-- create a base (hyper)table
CREATE TABLE measurements (
    generated_at timestamptz not null,
    value int4 not null,
    location_id int4 not null,
    PRIMARY KEY (location_id, generated_at)
);

SELECT create_hypertable (
               'measurements',
               'generated_at',
               chunk_time_interval => INTERVAL '2 days'
       );

-- add data
INSERT INTO measurements(generated_at, value, location_id)
SELECT generated_at, extract(EPOCH FROM generated_at)::int,location_id
FROM generate_series(1::int,2, 1) location_id,
     generate_series('2022-01-01T00:00:00.000UTC'::timestamptz,
                     CURRENT_TIMESTAMP at time zone 'UTC', '15 minutes') as generated_at;
;


-- Add a continuous aggregate on a continuous aggregate.
CREATE MATERIALIZED VIEW summary_day WITH (timescaledb.continuous)
AS SELECT
       time_bucket('1 day', m.generated_at, 'UTC') AS "time",
       m.location_id,
       count(*) AS count
   FROM measurements AS m
   GROUP BY m.location_id, (time_bucket('1 day', m.generated_at, 'UTC'));

CREATE MATERIALIZED VIEW summary_month WITH (timescaledb.continuous)
AS SELECT
       time_bucket('1 month', s.time, 'UTC') AS "time",
       s.location_id,
       sum(s.count)::integer AS count
   FROM summary_day AS s
   GROUP BY s.location_id, (time_bucket('1 month', s.time::timestamptz, 'UTC'))
WITH NO DATA;


-- This one fails
CALL refresh_continuous_aggregate('summary_month', '2022-01-01T00:00:00.000UTC'::timestamptz, '2022-02-02T00:00:00.000UTC'::timestamptz);
-- This one works adding 5 days
CALL refresh_continuous_aggregate('summary_month','2022-01-01T00:00:00.000UTC'::timestamptz,'2022-02-06T00:00:00.000UTC'::timestamptz);
--but this one fails
CALL refresh_continuous_aggregate('summary_month','2022-02-01T00:00:00.000UTC'::timestamptz,'2022-03-06T00:00:00.000UTC'::timestamptz);

-- no data from 2022-01-01
SELECT *
FROM summary_month
WHERE time < '2022-02-01';
-- nor in the underlying table
SELECT *
FROM _timescaledb_internal._materialized_hypertable_259
WHERE time < '2022-02-01';
-- increasing the range to about
CALL refresh_continuous_aggregate('summary_month','2021-12-02T00:00:00.000UTC'::timestamptz,'2022-02-06T00:00:00.000UTC'::timestamptz);
-- produces data
SELECT *
FROM summary_month
WHERE time < '2022-02-01';`

**Comment by linusback** [comment]
Update on this. 
I tried to change the bucket_width in _timescaledb_catalog.continuous_aggs_bucket_function 
for the underlying hypertable of summary_month in the previous example. 

```
UPDATE _timescaledb_catalog.continuous_aggs_bucket_function
SET bucket_width='1 month'
WHERE mat_hypertable_id=<id>

-- this now works 
CALL refresh_continuous_aggregate('summary_month', '2022-01-01T00:00:00.000UTC'::timestamptz, '2022-02-01T00:00:00.000UTC'::timestamptz);
```

It was set to 30 days instead of 1 month, probably for some underlying reason i don't understand. 
Worth noting is that if you do a normal CA directly on meassurement for 1 month then the bucket_width is set to '1 month' not '30 days'.

**Comment by fabriziomello** [comment]
@linusback this was fixed by #5860 and released in [2.12.0](https://github.com/timescale/timescaledb/releases/tag/2.12.0). I recommend you update to 2.12.2 (the last of 2.12.x series) to have this fix.
